src/main/java/core/basesyntax/dao/CarDaoJdbcImpl.java

```diff
@@ -0,0 +1,196 @@
+                 PreparedStatement preparedStatement = connection.prepareStatement(getByIdQuery)) {
+            preparedStatement.setLong(1, id);
+            ResultSet resultSet = preparedStatement.executeQuery();
+            if (resultSet.next()) {
+                return Optional.of(getCar(resultSet));
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Cannot get driver by id " + id, e);
+        }
+        return Optional.empty();
+    }
+
+    @Override
+    public List<Car> getAll() {
+        String getAllQuery = "SELECT * FROM cars c"
+                + " INNER JOIN manufacturers m ON c.manufacturer_id = m.id"
+                + " WHERE c.deleted = false";
+        List<Car> cars = new ArrayList<>();
+        try (Connection connection = ConnectionUtil.getConnection();
+                 PreparedStatement preparedStatement = connection.prepareStatement(getAllQuery)) {
+            ResultSet resultSet = preparedStatement.executeQuery();
+            while (resultSet.next()) {
+                cars.add(getCar(resultSet));
+            }
+            return cars;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Cannot get any cars", e);
+        }
+    }
+
+    @Override
+    public Car update(Car car) {
+        String updateQuery = "UPDATE cars SET manufacturer_id = ?,"
+                + " model = ? WHERE id = ? AND deleted = false";
+        String deleteCarsDrivers = "DELETE FROM cars_drivers WHERE car_id = ?;";
+        String insertCarsDrivers = "INSERT INTO cars_drivers (car_id, driver_id) VALUES(?, ?);";
+        try (Connection connection = ConnectionUtil.getConnection();
+                 PreparedStatement updatePreparedStatement = connection
+                         .prepareStatement(updateQuery);
+                 PreparedStatement deletePreparedStatement = connection
+                         .prepareStatement(deleteCarsDrivers);
+                 PreparedStatement insertPreparedStatement = connection
+                         .prepareStatement(insertCarsDrivers)) {
+            updatePreparedStatement.setLong(1, car.getManufacturer().getId());
+            updatePreparedStatement.setString(2, car.getModel());
+            updatePreparedStatement.setLong(3, car.getId());
+            updatePreparedStatement.executeUpdate();
+            deletePreparedStatement.setLong(1, car.getId());
+            deletePreparedStatement.executeUpdate();
+            insertPreparedStatement.setLong(1, car.getId());
+            for (Driver driver : car.getDrivers()) {
+                insertPreparedStatement.setLong(2, driver.getId());
+                insertPreparedStatement.executeUpdate();
+            }
+            return car;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Cannot update car " + car, e);
+        }
+    }
+
+    @Override
+    public boolean delete(Long id) {
```

Comment:
you must delere relations in the cars_drivers for the car with this id

Reply:
I do this in CarServiceImpl